### PR TITLE
Match and link ogScreen.cpp

### PR DESCRIPTION
**Pull request by HeartPiece44**
Links ogScreen.cpp and fixes a sizeable amount of JSU headers courtesy of the Twilight Princess repo. J2DPane::getTagName() should be weak, but refuses to generate, so it's currently global. Might be a future issue, unsure.

### CODE REVIEW

File: include/JSystem/J2D/J2DPane.h [hunk 1]
```diff
@@ -115,7 +116,7 @@ struct J2DPane {
 	J2DPane* getFirstChildPane();
 	J2DPane* getNextChildPane();
 	J2DPane* getParentPane();
-	JSUTree<J2DPane>* getPaneTree();
+	JSUTree<J2DPane>* getPaneTree() { return &m_tree; }
```
Comment (EpochFlame) [inline include/JSystem/J2D/J2DPane.h:119]: definitely resolve this weak function in the future

File: include/og/Screen/ogScreen.h [hunk 2]
```diff
@@ -109,6 +110,10 @@ struct DispMemberBase {
 	DispMemberBase* m_subMember; // _04
 };
 
+extern const char* PikiIconTextureName[19];
```
Comment (EpochFlame) [inline include/og/Screen/ogScreen.h:113]: no need to extern in headers, as that is implicit in the declaration iirc

File: src/plugProjectOgawaU/ogScreen.cpp [hunk 3]
```diff
@@ -1,1880 +1,461 @@
...
-fmadds   f1, f2, f1, f0
-addi     r1, r1, 0x20
-blr
-	*/
+	_00 += 30.0f * (m_speed * sys->m_secondsPerFrame);
+	if (_00 > TAU) {
+		_00 = _00 - TAU;
+	}
+	float factor = _08 * (1.0f + pikmin2_sinf(_00));
+	float scale  = 0.5f;
```
Comment (EpochFlame) [inline src/plugProjectOgawaU/ogScreen.cpp:46]: might this be a simple division by two?

**Review by EpochFlame — CHANGES_REQUESTED** [review]
see the comments I left and investigate them, please

File: src/plugProjectOgawaU/ogScreen.cpp (hunk 3 above)
Comment (jonathonbrady) [reply src/plugProjectOgawaU/ogScreen.cpp:46]: Looks like it: https://decomp.me/scratch/kNKXB

File: src/plugProjectOgawaU/ogScreen.cpp [hunk 4]
```diff
@@ -1,1880 +1,461 @@
...
-lwz      r28, 0x20(r1)
-mtlr     r0
-addi     r1, r1, 0x40
-blr
-	*/
+	u16 keta = 1;
+	for (int i = 1; i < 10; i++) {
+		if (p1 >= pow(10.0, (double)i)) {
+			short j = i + 1;
+			keta    = j;
```
Comment (jonathonbrady) [inline src/plugProjectOgawaU/ogScreen.cpp:251]: Any reason to declare `j` here? `keta = i + 1;` has the same codegen: https://decomp.me/scratch/45TKK

File: src/plugProjectOgawaU/ogScreen.cpp [hunk 5]
```diff
@@ -1,1880 +1,461 @@
...
+ * Size:	0000B0
  */
-void centeringPane(J2DPane*, bool, bool)
+void calcGlbCenter(J2DPane* pane, Vector2f* center)
 {
-	// UNUSED FUNCTION
+	Vector3f vec1 = pane->getGlbVtx(0);
+	Vector3f vec2 = pane->getGlbVtx(3);
+	center->x     = (vec1.x + vec2.x) / 2.f;
+	center->y     = (vec1.y + vec2.y) / 2.f;
```
Comment (jonathonbrady) [inline src/plugProjectOgawaU/ogScreen.cpp:184]: Nit: I'd prefer `2` over `2.f` here since we don't use this syntax anywhere else

Comment (EpochFlame) [reply src/plugProjectOgawaU/ogScreen.cpp:184]: agreed

File: include/og/Screen/ogScreen.h (hunk 2 above)
Comment (HeartPiece44) [reply include/og/Screen/ogScreen.h:113]: I was under the impression that externing in headers was just useful for generating headers for other files (for decomp.me, etc), and not putting extern would gen the instance in the first file linked using the header

File: src/plugProjectOgawaU/ogScreen.cpp (hunk 3 above)
Comment (HeartPiece44) [reply src/plugProjectOgawaU/ogScreen.cpp:46]: will update o7

File: src/plugProjectOgawaU/ogScreen.cpp (hunk 4 above)
Comment (HeartPiece44) [reply src/plugProjectOgawaU/ogScreen.cpp:251]: probably a hangover from sorting out type issues - will remove it

File: src/plugProjectOgawaU/ogScreen.cpp (hunk 5 above)
Comment (HeartPiece44) [reply src/plugProjectOgawaU/ogScreen.cpp:184]: yeah this was a copy-paste from encounter's solution - I'll admit I didn't investigate whether it was required or not for it to match. will double check and update accordingly

File: include/JSystem/J2D/J2DPane.h (hunk 1 above)
Comment (HeartPiece44) [reply include/JSystem/J2D/J2DPane.h:119]: I assume this is actually about `J2DPane::getTagName` which is currently global when it should be weak - will need to revisit with more files decomp'd, agreed.

**Review by EpochFlame — APPROVED** [review]

**Review by EpochFlame — APPROVED** [review]
lgtm

**Pull request merged**